src/utils/storage.py

```diff
@@ -28,6 +33,23 @@ def remove_file(self, file_path):
         """
         Remove a blob from Azure Blob Storage
         """
+        if file_path.startswith("https://"):
+            parsed_url = parse.urlparse(file_path)
+            file_path = parsed_url.path.lstrip("/")
+
+        container_name = self.container_client.container_name
+
+        if file_path.startswith(f"{container_name}/"):
+            file_path = file_path[len(f"{container_name}/") :]
+
```

Comment:
I don't understand the purpose of these. Doesn't his function will always receive the full url (with https)

Reply:
Explanation of the Code
This code ensures that the correct internal path of a blob is used when attempting to delete it from Azure Blob Storage. Below is a detailed explanation of why each part of the logic is necessary:

Extracting the Path from the URL:

if file_path.startswith("https://"):

    parsed_url = parse.urlparse(file_path)
    file_path = parsed_url.path.lstrip("/")

Why:
If the function receives a full URL (e.g., https://staccatalog.blob.core.windows.net/cog-test/LossPersistence/2016-2021.tif), it needs to extract the path component to use it with Azure's get_blob_client().

Example: From the URL, it extracts the path:

cog-test/LossPersistence/2016-2021.tif.

lstrip("/") ensures any leading slashes are removed, leaving the path clean.

---------------
Removing the Container Name from the Path:

container_name = self.container_client.container_name

if file_path.startswith(f"{container_name}/"):
    file_path = file_path[len(f"{container_name}/") :]

Why:

Azure Blob Storage expects only the internal path of the blob (relative to the container) when accessing it. If the extracted path includes the container name, it must be removed.

Example:
If the path is cog-test/LossPersistence/2016-2021.tif and the container name is cog-test, the path is adjusted to:
LossPersistence/2016-2021.tif.

Reply:
yeap, thanks, i was confused with the two ifs thinking in the second as an `elif` :sweat_smile: :see_no_evil: